### PR TITLE
Update dependencies and tools

### PR DESCRIPTION
**Pull request by tksm**
This PR updates the dependencies for the release of [Kubernetes v1.28.0](https://github.com/kubernetes/kubernetes/releases/tag/v1.28.0). I checked the latest versions of the tool on the following URLs.

- https://go.dev/doc/devel/release
- https://github.com/goreleaser/goreleaser/releases
- https://github.com/golangci/golangci-lint/releases
- https://github.com/kubernetes-sigs/krew/releases
- https://github.com/t0yv0/goreleaser-filter/releases

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +54,7 @@ verify-readme:
 
 .PHONY: validate-krew-manifest
 validate-krew-manifest: $(VALIDATE_KREW_MAIFEST)
-	$(VALIDATE_KREW_MAIFEST) -manifest dist/stern.yaml -skip-install
+	$(VALIDATE_KREW_MAIFEST) -manifest dist/krew/stern.yaml -skip-install
```
Comment (tksm) [inline Makefile:57]: :memo: The manifest path was changed in https://github.com/goreleaser/goreleaser/commit/76ce66c060341cd55983e68624c17f4e46377619.

File: cmd/cmd_test.go [hunk 2]
```diff
@@ -521,7 +521,7 @@ func TestOptionsSternConfig(t *testing.T) {
 				c.AllNamespaces = true
 				c.LabelSelector = labelSelector
 				c.FieldSelector = fieldSelector
-				c.TailLines = pointer.Int64(10)
+				c.TailLines = ptr.To[int64](10)
```
Comment (tksm) [inline cmd/cmd_test.go:524]: :memo: The `pointer` package was deprecated in https://github.com/kubernetes/utils/commit/dc1a5d8ab3834835ab2294895323df879d7192ba.

**Review by rkmathi — APPROVED** [review]
LGTM 🎉

**Review by floryut — APPROVED** [review]
@tksm Thank you 👍

**Pull request merged**